packages/flutter/lib/src/widgets/sliver_tree.dart

```diff
@@ -0,0 +1,954 @@
+        'SliverTree.',
+      ),
+      ErrorDescription(
+        'No SliverTree ancestor could be found starting from the context that '
+        'was passed to TreeController.of(). '
+        'This usually happens when the context provided is from the same '
+        'StatefulWidget as that whose build function actually creates the '
+        'SliverTree widget being sought.',
+      ),
+      ErrorHint(
+        'There are several ways to avoid this problem. The simplest is to use '
+        'a Builder to get a context that is "under" the SliverTree.',
+      ),
+      ErrorHint(
+        'A more efficient solution is to split your build function into '
+        'several widgets. This introduces a new context from which you can '
+        'obtain the SliverTree. In this solution, you would have an outer '
+        'widget that creates the SliverTree populated by instances of your new '
+        'inner widgets, and then in these inner widgets you would use '
+        'TreeController.of().',
+      ),
+      context.describeElement('The context used was'),
+    ]);
+  }
+
+  /// Finds the [SliverTreeList] from the closest instance of this class that
+  /// encloses the given context and returns its [SliverTreeController].
+  ///
+  /// If no [SliverTreeList] encloses the given context then return null.
+  /// To throw an exception instead, use [of] instead of this function.
+  ///
+  /// See also:
+  ///
+  ///  * [of], a similar function to this one that throws if no [SliverTreeList]
+  ///    encloses the given context. Also includes some sample code in its
+  ///    documentation.
+  static SliverTreeController? maybeOf(BuildContext context) {
+    return context.findAncestorStateOfType<_SliverTreeListState<dynamic>>()?.controller;
+  }
+}
+
+int _kDefaultSemanticIndexCallback(Widget _, int localIndex) => localIndex;
+
+/// A sliver for lazily displaying [SliverTreeNode]s that expand and collapse in
+/// a vertically scrolling [Viewport].
+///
+/// The rows of the tree are laid out on demand, using
+/// [SliverTreeList.treeNodeBuilder]. This will only be called for the nodes
+/// that are visible, or within the [Viewport.cacheExtent].
+///
+/// Only [Viewport]s that scroll with and axis direction of [AxisDirection.down]
+/// can use SliverTree.
+///
+///{@tool dartpad}
+/// This example uses a [SliverTreeList] to display nodes, highlighting nodes as
+/// they are selected.
+///
+/// ** See code in examples/api/lib/widgets/sliver/sliver_tree.0.dart **
+/// {@end-tool}
+///
+/// {@tool dartpad}
+/// This example shows a highly customized [SliverTreeList] configured to
+/// [SliverTreeIndentationType.none]. This allows the indentation to be handled
+/// by the developer in [SliverTreeList.treeNodeBuilder], where a decoration is
+/// used to fill the indented space.
+///
+/// ** See code in examples/api/lib/widgets/sliver/sliver_tree.1.dart **
+/// {@end-tool}
+class SliverTreeList<T> extends StatefulWidget {
```

Comment:
I do not love this name, but SliverTree and SliverAnimatedTree are taken. Open to other ideas. I think this works because the tree itself is flattened to a list of active nodes. 
And it subclasses SliverVariedExtentList. So.. it's list-y.